subprojects/core/src/main/java/org/gradle/api/internal/classpath/DefaultModuleRegistry.java

```diff
@@ -179,6 +179,7 @@ private void findImplementationClasspath(String name, Collection<File> implement
         matcher.matches();
         String projectDirName = matcher.group(1);
         String projectName = toCamelCase(projectDirName);
+        suffixes.add(("/" + projectDirName + "/out/production/classes").replace('/', File.separatorChar));
```

Comment:
:o: I noticed this function creates a lot of temporaries for each module search, maybe we could improve on that?

Another thing is that it feels very imprecise with this search path list. It would be nice if the code could decide once where to look for modules and then always use the same strategy.